File: Hashing/insert.java
public class insert {
    public static void main(String[] args){
        int a[] = {14,9,15,12,6,8,13};
        int n = a.length;
        // System.out.print(n);
        for(int i = 0; i <n-1 ;i++){
            int j = i;
            while(j>0 && a[j-1]>a[j]){
                int temp = a[j];
                a[j-1] = a[j];
                a[j] = temp;
                j--;
            } 
        }
        for(int i = 0; i <n ;i++){
            System.out.println(a[i]);
        }
    }
}

File: insert.py
# arr = [4,1,3,5,6]
# n = len(arr)
# # print(n)
# def insertSorting(arr):
#     for i in range(n-1):
#         for j in range(n-1):
#             if(arr[i]<arr[j]):
#                 temp = arr[i]
#                 arr[i] = arr[j]
#                 arr[j] = temp
#     print(arr)

# insertSorting(arr)

def insertSorting(arr):
    n = len(arr)  # Ensure n is defined as the length of the array
    for i in range(n-1):
        for j in range(n-1):
            if(arr[i] < arr[j]):
                temp = arr[i]
                arr[i] = arr[j]
                arr[j] = temp
    return arr  # Use return instead of print

# Example usage
arr = [5, 3, 8, 6, 2]
sorted_arr = insertSorting(arr)
print(sorted_arr)  # This will print the returned sorted array
